packages/vite/src/node/server/send.ts

```diff
@@ -56,10 +58,20 @@ export function send(
   }
 
   // inject source map reference
-  if (map && map.mappings) {
+  if (map && 'version' in map && map.mappings) {
     if (type === 'js' || type === 'css') {
       content = getCodeWithSourcemap(type, content.toString(), map)
     }
+  } else {
+    if (type === 'js' && (!map || map.mappings !== '')) {
+      const urlWithoutTimestamp = removeTimestampQuery(req.url!)
+      const ms = new MagicString(content.toString())
+      content = getCodeWithSourcemap(
+        type,
+        content.toString(),
+        ms.generateMap({ source: urlWithoutTimestamp, hires: 'boundary' }),
+      )
+    }
```

Comment:
This part is the main part of the change. Other changes are to pass `{ mappings: '' }` as-is to this part.